src/Chainweb/Pact/PactService.hs

```diff
@@ -823,13 +827,14 @@ execHistoricalLookup
     -> PactServiceM logger tbl (Maybe (P.TxLog P.RowData))
 execHistoricalLookup bh d k = pactLabel "execHistoricalLookup" $ do
   !cp <- getCheckpointer
+  -- TODO: pass parent header to limit the visible range
```

Comment:
Hm, I think it doesn't make sense.

The limitation would make sense only in situations when the parent header got out of sync and normally we don't need to limit these queries as we don't do any rewinds here.